### PR TITLE
Refs [TASK][34043] Handle event Detail Meal Screen

### PR DESCRIPTION
**Pull request by phucsaker99**
## Related Tickets
- [#Task 34043: Handle event Detail Meal Screen](https://edu-redmine.sun-asterisk.vn/issues/34043)
## WHAT
- Bind Data : get data pour into textview, imageview
- Item click imageview: zoom in and zoom out image
- Item click imageButton: add/remove meal to my favorite list
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt [hunk 1]
```diff
@@ -1,15 +1,156 @@
...
     override val layoutResource: Int
         get() = R.layout.fragment_meal_detail
 
-    override fun initComponents() {
+    override fun onStop() {
+        super.onStop()
+        when (buttonFavorite.visibility) {
+            View.VISIBLE -> meal?.let { presenter?.insertMealFavorite(it) }
+            else -> meal?.let { presenter?.deleteMealFavorite(it.id) }
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt:26]: sao lại viết vào onStop thế

File: app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt [hunk 2]
```diff
@@ -1,15 +1,156 @@
...
+            loadingDialog = LoadingDialog(it)
+        }
+    }
+
+    private fun initInputData() {
+        meal = arguments?.getParcelable<Meal>(BUNDLE_MEAL_DETAIL)?.let { meal ->
+            imageMealDetail.loadImageCircle(meal.image)
+            textMealName.text = meal.name
+            meal
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt]: dùng apply

File: app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private var scaleGestureDetector: ScaleGestureDetector? = null
+    private var mScaleFactor = 1.0f
+
+    override val layoutResource get() = R.layout.fragment_image_zoom
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun setupViews() {
+        arguments?.let {
+            val image = it.getString(IMAGE_ZOOM)
+            Glide.with(requireContext()).load(image).into(imageZoom)
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt]: không dùng requireContext() 
check null context

File: app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import androidx.core.os.bundleOf
+import com.bumptech.glide.Glide
+import com.example.yummy.R
+import com.example.yummy.base.BaseFragment
+import com.example.yummy.utlis.removeFragment
+import kotlinx.android.synthetic.main.fragment_image_zoom.*
+
+class ZoomImageFragment : BaseFragment() {
+    private var scaleGestureDetector: ScaleGestureDetector? = null
+    private var mScaleFactor = 1.0f
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt]:
```suggestion
    private var scaleFactor = 1.0f
```

File: app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    private inner class ScaleListener : ScaleGestureDetector.SimpleOnScaleGestureListener() {
+        override fun onScale(scaleGestureDetector: ScaleGestureDetector): Boolean {
+            mScaleFactor *= scaleGestureDetector.scaleFactor
+            mScaleFactor = 0.1f.coerceAtLeast(mScaleFactor.coerceAtMost(10.0f))
+            imageZoom.scaleX = mScaleFactor
+            imageZoom.scaleY = mScaleFactor
+            return true
+        }
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/mealdetail/ZoomImageFragment.kt]: Để cuối class

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 6]
```diff
@@ -35,7 +34,15 @@ class MealListFragment : BaseFragment(), MealListContract.View {
     }
 
     override fun initActions() {
-        buttonBack.setOnClickListener { parentFragmentManager.removeFragment(R.id.frameMain, this) }
+        buttonBack.apply {
+            increaseHitArea(20f)
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]: hard code

File: app/src/main/res/layout/fragment_image_zoom.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_image_zoom.xml]: Dùng constraint layout

File: app/src/main/res/layout/fragment_image_zoom.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_centerHorizontal="true"
+        android:layout_gravity="center"
+        android:contentDescription="@null" />
+
+    <ImageButton
+        android:id="@+id/imageCancel"
+        android:layout_width="40dp"
+        android:layout_height="40dp"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_image_zoom.xml]: hard code

File: app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt (hunk 1 above)
Comment (phucsaker99) [reply app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt:26]: e tránh người dùng click nhiều lần vào buttonFavorite làm check local data nhiều lần, khi thoát màn hình ms kiểm tra xem nút Favorite đó có khác biệt ko ms check local data

File: app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt (hunk 2 above)
Comment (phucsaker99) [reply app/src/main/java/com/example/yummy/ui/mealdetail/MealDetailFragment.kt]: vâng a

File: app/src/main/res/layout/fragment_image_zoom.xml (hunk 7 above)
Comment (phucsaker99) [reply app/src/main/res/layout/fragment_image_zoom.xml]: vâng a

**Pull request merged**